scripts/home.js

```diff
@@ -0,0 +1,185 @@
+        category: "Beauty",
+        details: "Some details"
+    },
+    {
+        id: generateRandomNumber(),
+        name: "Dress",
+        price: 50, 
+        imageURL: "https://cdn.shopify.com/s/files/1/0011/9783/4252/products/20_375a8763-f5d7-4184-a352-4523ef713733_900x.jpg?v=1576267132", 
+        category: "Fashion",
+        details: "Some details"
+    },
+    {
+        id: generateRandomNumber(),
+        name: "Lip Gloss",
+        price: 10, 
+        imageURL: "https://www.sephora.com/productimages/sku/s1925965-main-zoom.jpg", 
+        category: "Beauty",
+        details: "Some details"
+    },
+    {
+        id: generateRandomNumber(),
+        name: "iphone 12",
+        price: 1300,
+        imageURL: "https://www.telstra.com.au/content/dam/tcom/lego/2020/plans-devices/mobiles/iphone-12-pro-max/shared-iphone-12-pro-max-pacific-blue-01-900x1200.jpg",
+        category: "Technology",
+        details: "Some details"
+    },
+    {
+        id: generateRandomNumber(), 
+        name: "Shoes",
+        price: 160,
+        imageURL:"https://cdn.fleetfeet.com/products/1012A714_020_SR_RT_PNG_1280x1280-JPG.jpg/width:720-height:540-fit:trim/1012A714_020_SR_RT_PNG_1280x1280-JPG.jpg?signature=205cac59",
+        category: "Fashion",
+        details: "Some details"
+    },
+    {
+        id: generateRandomNumber(), 
+        name: "Jumping Rope",
+        price: 20,
+        imageURL:"https://www.dragonsports.eu/333581-verylarge_default/nylon-jumping-rope-grip-foam-adi1102-adidas.jpg",
+        category: "Fitness",
+        details: "Some details"
+    }
+
+
+]; 
+console.log(typeof products[0].id);
+
+
+// Hamburger Menu Animation
+const burgerMenu = document.querySelector(".hamburger-menu");
+const menuContent = document.querySelector(".menu-content");
+
+burgerMenu.addEventListener("click",()=>{
+    burgerMenu.classList.toggle("open");
+    if(menuContent.classList.contains("show")){
+        menuContent.classList.remove("show"); 
+        menuContent.classList.add("hide"); 
+    }else{
+        menuContent.classList.remove("hide"); 
+        menuContent.classList.add("show");
+    }
+}); 
+
+
+// Small screen search icon
+const searchBox = document.querySelector(".search-box");
+const searchIcon = document.querySelector(".search-icon");
+searchInputField =  document.querySelector(".search-box input[type='text']");
+searchIcon.addEventListener("click",()=>{
+    if(searchBox.classList.contains("hide-search-box")){
+        searchBox.classList.remove("hide-search-box");
+        searchInputField.value = "";
+    }else{
+        searchBox.classList.add("hide-search-box");
+    }
+});  
+
+// prevent page from reload when the search form get submitted 
+document.querySelector(".search-box").addEventListener("submit",event=>{
+    event.preventDefault(); 
+    searchInputField.value = "";
+});
+
+
+// Remove the menu and search box
+// if screen width > 550
+window.onresize = ()=>{
+    if(window.innerWidth > 550){
```

Comment:
If you want, you could accomplish this same thing purely in css using media queries:
https://css-tricks.com/css-media-queries/

This is how most companies are building their responsive css.

Reply:
Thank you